### PR TITLE
feat(java): add warning-level logger rule

### LINKED ISSUE #290: Add warning-level CWE-532 rule for Java
The current CWE-532 rule for Java concerns logging Data Types. We should also catch cases of logging any dynamic input, as this could also be a security leak, for example:

```java
logger.error("Logging a variable: " + someVar);
```

This new CWE-532 rule should have a warning level severity

### PR DESCRIPTION
**Pull request by elsapet**
## Description
Add warning-level logger rule for any logger instances that are not pure strings

Closes #290 

## Checklist

- [ ] I've added a snapshot that shows my rule works as expected.
- [ ] My rule has adequate metadata to explain its use.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: rules/java/lang/logger.yml [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-  - pattern: |
-      $<LOG>.$<METHOD>($<...>$<DATA_TYPE>$<...>)
+  - pattern: $<LOG>.$<METHOD>($<LOG_MESSAGE>)
     filters:
-      - variable: DATA_TYPE
-        detection: java_shared_lang_datatype
-        scope: result
+      - not:
+          variable: LOG_MESSAGE
+          regex: \A("|').*("|')\z
```
Comment (cfabianski) [inline rules/java/lang/logger.yml:8]:
```suggestion
          detection: string_literal
```

Comment (elsapet) [reply rules/java/lang/logger.yml:8]: Updating it 😎

File: rules/java/lang/logger.yml [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-      $<LOG>.$<METHOD>($<...>$<DATA_TYPE>$<...>)
+  - pattern: $<LOG>.$<METHOD>($<INPUT>)
     filters:
-      - variable: DATA_TYPE
-        detection: java_shared_lang_datatype
-        scope: result
+      - not:
+          variable: INPUT
+          detection: string_literal
+          scope: result
```
Comment (cfabianski) [inline rules/java/lang/logger.yml:10]:
```suggestion
          scope: cursor
```

**Comment by cfabianski** [comment]
Requires some snapshot updates :)

**Pull request merged**